ui/plugin-system/src/runtime/DataQueriesProvider/model.ts

```diff
@@ -11,32 +11,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Definition, TimeSeriesData, UnknownSpec } from '@perses-dev/core';
+import { Definition, QueryDefinition, QueryType, UnknownSpec } from '@perses-dev/core';
+import { QueryDataType } from '@perses-dev/core';
+import { UseQueryResult } from '@tanstack/react-query';
 
 type QueryOptions = Record<string, unknown>;
 
-interface DataQueriesDefinitions<QueryPluginDefinition> {
-  definitions: QueryPluginDefinition[];
+export interface DataQueriesDefinition extends Definition<UnknownSpec> {
+  /**
+   * @default 'TimeSeriesQuery'
+   */
+  type?: keyof QueryType;
```

Comment:
We need a new prop `type` here since it is now capable of supporting other query types

Reply:
If we made the data query type unique across all types, it would be cool to infer this.
eg. You could look up that `PrometheusTimeSeriesQuery` was a `TimeSeriesQuery` without explicitly saying so.

Reply:
Would that be a problem for our BE validation though? One of the goals is complete static validation of the entire schema in CI for GitOps use cases

Reply:
> Would that be a problem for our BE validation though? One of the goals is complete static validation of the entire schema in CI for GitOps use cases

I don't believe so. Can you give an example of where the issue would be?

Reply:
> You could look up that PrometheusTimeSeriesQuery was a TimeSeriesQuery without explicitly saying so.

I misunderstood, I thought you were referring to how we've talked about someday trying to remove the [nesting](https://github.com/perses/perses/blob/v0.38.0/dev/data/dashboard.json#L163-L174) of having PrometheusTimeSeriesQuery inside of TimeSeriesQuery in our dashboard schema

For that topic, I wasn't sure if the BE relies on the nesting, but I'd have to dig into the implementation more to understand: [internal/api/shared/schemas/schemas.go#L76-L98](https://github.com/perses/perses/blob/a999795fdd1c2ce9ee76db7e429465012d7d6623/internal/api/shared/schemas/schemas.go#L76-L98)

Reply:
I think it does right now, but I don't think there is a strong reason why it needs to. 

For this change however, we are focusing on the framework code rather than the dashboard spec translation which can be slightly different.